src/rustdoc_gen.rs

```diff
@@ -123,11 +182,48 @@ pub(crate) enum CrateType<'a> {
     },
 }
 
+#[derive(Debug, Clone, Hash)]
+pub(crate) struct FeatureConfig {
+    pub(crate) base_features: BaseFeatures,
+    pub(crate) extra_features: Vec<String>,
+    pub(crate) ignore_non_existing: bool,
+}
+
+#[derive(Debug, Clone, Hash)]
+pub(crate) enum BaseFeatures {
+    All,
+    Default,
+    Heuristic,
+    None,
+}
```

Comment:
Does this "base" here refer to "baseline", or is it completely unrelated? Can we come up with a term that won't be confusing?

Reply:
No, I meant "foundation". I have trouble coming up with a better name. Maybe `StartingFeatures`? That seems awkward as well.

Reply:
How about `FeatureGroup`?